src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,1107 @@
+  integer, private :: i_avl_som  ! index of available (aka active) SOM
+  integer, private :: i_str_lit  ! index of structural litter pool
+  integer, private :: i_cop_mic  ! index of copiotrophic microbial pool
+  integer, private :: i_oli_mic  ! index of oligotrophic microbial pool
+  integer, private :: i_l1m1  ! indices of pool transfers needed in whole module
+  integer, private :: i_l1m2
+  integer, private :: i_l2m1
+  integer, private :: i_l2m2
+  integer, private :: i_s1m1
+  integer, private :: i_s1m2
+  integer, private :: i_m1s1
+  integer, private :: i_m1s2
+  integer, private :: i_m2s1
+  integer, private :: i_m2s2
+
+  type, private :: params_type
+     ! Respiration fraction cwd --> structural litter (keep comment sep for dif)
+     real(r8):: rf_cwdl2_bgc 
+     real(r8):: tau_cwd_bgc   ! corrected fragmentation rate constant CWD, century leaves wood decomposition rates open, within range of 0 - 0.5 yr^-1 (1/0.3) (1/yr)
+     real(r8) :: cwd_flig_bgc !
+     real(r8) :: minpsi_bgc   !minimum soil water potential for heterotrophic resp
+     real(r8) :: maxpsi_bgc   !maximum soil water potential for heterotrophic resp
+
+     ! TODO Need for spin-ups?
+     real(r8) :: initial_Cstocks_depth      ! Soil depth for initial Carbon stocks for a cold-start
+     real(r8), allocatable :: initial_Cstocks(:)  ! Initial Carbon stocks for a cold-start
+     real(r8), allocatable :: mge(:)  ! Microbial growth efficiency (mg/mg)
+     real(r8), allocatable :: vmod(:)  ! 
+     real(r8), allocatable :: vslope(:)  ! 
+     real(r8), allocatable :: vint(:)  ! 
+     real(r8), allocatable :: kmod(:)  ! 
+     real(r8), allocatable :: kslope(:)  ! 
+     real(r8), allocatable :: kint(:)  ! 
+     
+  end type params_type
+  !
+  type(params_type), private :: params_inst
+
+  character(len=*), parameter, private :: sourcefile = &
+       __FILE__
+
+  !-----------------------------------------------------------------------
+
+contains
+
+  !-----------------------------------------------------------------------
+  subroutine readParams ( ncid )
+    !
+    ! !DESCRIPTION:
+    !
+    ! !USES:
+    use ncdio_pio    , only: file_desc_t,ncd_io
+    !
+    ! !ARGUMENTS:
+    type(file_desc_t),intent(inout) :: ncid   ! pio netCDF file id
+    !
+    ! !LOCAL VARIABLES:
+    character(len=32)  :: subname = 'readParams'
+    character(len=100) :: errCode = 'Error reading MIMICS params '
+    logical            :: readv   ! has variable been read in or not
+    real(r8)           :: tempr   ! temporary to read in constant
+    character(len=100) :: tString ! temp. var for reading
+    !-----------------------------------------------------------------------
+
+    ! Read off of netcdf file
+    ! TODO Add new params here and in the params file.
+    ! TODO Some may need _bgc added/removed here and in the params file
+    ! TODO Read MIMICS-specific params here & shared ones (eg *_bgc) as:
```

Comment:
may be nicer to put mimics_ as prefix rather than suffix to gather all the params together in the params file

get approval from Bill/Erik

Reply:
moved comment to the code